readthedocs/projects/tasks.py

```diff
@@ -1324,17 +1324,13 @@ def _manage_imported_files(version, path, commit):
     )
     # Keep the objects into memory to send it to signal
     instance_list = list(delete_queryset)
-    # Safely delete from database
-    delete_queryset.delete()
     # Always pass the list of instance, not queryset.
+    # These objects must exist though,
+    # because the task will query the DB for the objects before deleting
     bulk_post_delete.send(sender=HTMLFile, instance_list=instance_list)
+    # Safely delete from database
+    delete_queryset.delete()
 
-    # Delete ImportedFiles from previous versions
-    (
```

Comment:
All `HTMLFile` are `ImportedFile`, but not all `ImportedFile` are `HTMLFile`. So we should delete the `ImportedFile` also. The `HTMLFile` was deleted above for dispatching signals as `HTMLFile` is a proxy model.